nntrainer/tensor/var_grad.h

```diff
@@ -101,7 +101,12 @@ class Var_Grad {
    */
   explicit Var_Grad(Tensor *v, Tensor *g) :
     var(std::shared_ptr<Tensor>(v, [](void *) {})),
-    grad(std::shared_ptr<Tensor>(g, [](void *) {})) {}
+    grad(std::shared_ptr<Tensor>(g, [](void *) {})) {
+    if (!v)
+      var = std::make_shared<Tensor>();
+    if (!g)
+      grad = std::make_shared<Tensor>();
```

Comment:
Are we making one if not present?
How about we let it null instead now? (maybe later??)
 Related to https://github.com/nnstreamer/nntrainer/pull/1522#discussion_r717299885

Reply:
We can't leave it null because the dimensions in layers depends on RunContext which depends on its outputs and inputs. So, for now, vars and grads objects do exist, but they can be empty or unallocated.

There will be yet another cleanup once the rebase is done to clean up VarGrad various interfaces.

Reply:
Sure :)